src/components/application_manager/test/resumption/resume_ctrl_test.cc

```diff
@@ -642,6 +697,44 @@ TEST_F(ResumeCtrlTest, SetAppHMIState_HMIFull_WithPolicy_DevDisallowed) {
   EXPECT_FALSE(res);
 }
 
+TEST_F(ResumeCtrlTest, SaveAllApplications) {
+  utils::SharedPtr<application_manager_test::MockApplication> test_app =
+      ::utils::MakeShared<application_manager_test::MockApplication>();
+  EXPECT_CALL(*test_app, app_id()).WillRepeatedly(Return(kTestAppId_));
+
+  application_manager::ApplicationSet app_set;
+  app_set.insert(test_app);
+
+  DataAccessor<application_manager::ApplicationSet> accessor(app_set,
+                                                             app_set_lock_);
+
+  EXPECT_CALL(app_mngr_, applications())
+      .WillRepeatedly(Return(accessor));
+  EXPECT_CALL(*mock_storage, SaveApplication(Eq(test_app)));
+  res_ctrl->SaveAllApplications();
+
+  // Fix DataAccessor release and WinQt crash
+  Mock::VerifyAndClearExpectations(&app_mngr_);
```

Comment:
@LitvinenkoIra Please note, because it is better to use this (Mock::VerifyAndClearExpectations(&app_mngr_);) call in construtor and destructor or Setup and TearDown, please remove all other VerifyAndClearExpectations calls that are placed in some of the tests here.